Source/WebCore/page/Quirks.cpp

```diff
@@ -912,6 +912,11 @@ bool Quirks::shouldEnableEnumerateDeviceQuirk() const
 }
 #endif
 
+bool Quirks::shouldUnloadFrame() const
+{
+    return needsQuirks();
```

Comment:
This doesn't look right. `needsQuirks` just returns true unless the user has turned Quirks off entirely. This code means that we will always tell code checking `shouldUnloadFrame()` that we SHOULD unload the frame. That's not right.

Reply:
This should do something like:

`return needsQuirks() && m_quirksData.shouldUnloadHeavyFrames;`